Signal/src/util/DateUtil.m

```diff
@@ -53,4 +53,36 @@ + (BOOL)dateIsToday:(NSDate *)date {
     return [self date:[NSDate date] isEqualToDateIgnoringTime:date];
 }
 
++ (NSString *)formatPastTimestampRelativeToNow:(uint64_t)pastTimestamp
+{
+    OWSCAssert(pastTimestamp > 0);
+
+    uint64_t nowTimestamp = [NSDate ows_millisecondTimeStamp];
+    if (pastTimestamp >= nowTimestamp) {
+        OWSCFail(@"%@ Unexpected timestamp", self.tag);
+        return NSLocalizedString(@"TIME_NOW", @"Indicates that the event happened now.");
+    }
+
+    NSDate *pastDate = [NSDate ows_dateWithMillisecondsSince1970:pastTimestamp];
+    if ([self dateIsToday:pastDate]) {
+        return [[self timeFormatter] stringFromDate:pastDate];
```

Comment:
nit: should we dispatchOnce all the formatters for perf?

Reply:
Huh, `NSDateFormatter` has been thread-safe since iOS 7.  TIL